eng/automation/generate.py

```diff
@@ -283,8 +283,10 @@ def sdk_automation_typespec_project(tsp_project: str, config: dict) -> dict:
     breaking: bool = False
     changelog = ""
     breaking_change_items = []
+    run_mode: str = config["runMode"] if "runMode" in config else None
 
-    verify_self_serve_parameters(api_version, sdk_release_type)
+    if run_mode == "release":
```

Comment:
We also want to gate on the local run scenario. Could you also include `local` value?

Reply:
In which scenario will it be `local`? How is the pipeline triggered?

Reply:
`local` means the local run scenario. Users run `spec-gen-sdk` tool to generate code locally. We also want to include the validation for this case.

Reply:
Got it, added.